src/sass_types/null.cpp

```diff
@@ -15,45 +13,49 @@ namespace SassTypes
     return singleton_instance;
   }
 
-  Handle<Function> Null::get_constructor() {
+  v8::Local<v8::Function> Null::get_constructor() {
+    Nan::EscapableHandleScope scope;
+    v8::Local<v8::Function> conslocal;
     if (constructor.IsEmpty()) {
-      Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+      v8::Local<v8::FunctionTemplate> tpl = Nan::New<v8::FunctionTemplate>(New);
 
-      tpl->SetClassName(NanNew("SassNull"));
+      tpl->SetClassName(Nan::New("SassNull").ToLocalChecked());
       tpl->InstanceTemplate()->SetInternalFieldCount(1);
 
-      NanAssignPersistent(constructor, tpl->GetFunction());
+      constructor.Reset(Nan::GetFunction(tpl).ToLocalChecked());
+      conslocal = Nan::New(constructor);
 
-      NanAssignPersistent(get_singleton().js_object, NanNew(constructor)->NewInstance());
-      NanSetInternalFieldPointer(NanNew(get_singleton().js_object), 0, &get_singleton());
-      NanNew(constructor)->Set(NanNew("NULL"), NanNew(get_singleton().js_object));
+      get_singleton().js_object.Reset(conslocal->NewInstance());
+      Nan::SetInternalFieldPointer(Nan::New(get_singleton().js_object), 0, &get_singleton());
+      Nan::Set(conslocal, Nan::New("NULL").ToLocalChecked(), Nan::New(get_singleton().js_object));
 
       constructor_locked = true;
+    } else {
+      conslocal = Nan::New(constructor);
     }
 
-    return NanNew(constructor);
+    return scope.Escape(conslocal);
   }
 
   Sass_Value* Null::get_sass_value() {
     return sass_make_null();
   }
 
-  Local<Object> Null::get_js_object() {
-    return NanNew(this->js_object);
+  v8::Local<v8::Object> Null::get_js_object() {
+    return Nan::New(this->js_object);
```

Comment:
I am confused :)

Let me follow the call chain:

[`CallbackBridge<>::operator()(std::vector<void*> argv)`](https://github.com/saper/node-sass/blob/a1395134907c6167a71487f5937869f7ca7d21b2/src/callback_bridge.h#L80-L91) calls `pre_process_args()` in two ways:

1) directly in the sync case - we run in the V8 thread, so we create `EscapableHandleScope` and call `pre_process_args()`.  (Note: Now I think `HandleScope` should be enough here).
2) indirectly in the async case - we schedule [`CallbackBridge<>::ReturnCallback`](https://github.com/saper/node-sass/blob/a1395134907c6167a71487f5937869f7ca7d21b2/src/callback_bridge.h#L120-L138) which is a `NAN_METHOD` and has implicit `HandleScope`.

One of those methods, [`CustomFunctionBridge::pre_process_args()`](https://github.com/saper/node-sass/blob/a1395134907c6167a71487f5937869f7ca7d21b2/src/custom_function_bridge.cpp#L14-L22), calls `get_js_object()` so I assume it is all in the scope opened as described above?

`get_js_object()` is also called from some types `New()` methods (all defined via `NAN_METHOD()`), where it is wrapped in their implicit `HandleScope`.

If the above is correct, I might not need `EscapableScope` in the `get_js_object()`, right? Can I use calling function `(Escapable)HandleScope` instead?